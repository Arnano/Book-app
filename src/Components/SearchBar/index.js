import React from 'react';
import PropTypes from 'prop-types';
import StyledSearchBar from './style';

import growth from '../../Style/assets/icons/growth.svg';

import Image from '../Image';
import Input from '../Input';
import Button from '../Button';

const SearchBar = ({ children, onChange, onClick, icon, value}) => 
  <StyledSearchBar>
    <Image classes='c-search-bar__icon' src={growth} alt='searchIcon' />
    <Input
      onChange={onChange}
      value={value} />
    <Button
      onClick={onClick} >
      {children}
    </Button>
  </StyledSearchBar>

  SearchBar.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func,
    onClick: PropTypes.func,
    children: PropTypes.node
  }

  export default SearchBar;
  