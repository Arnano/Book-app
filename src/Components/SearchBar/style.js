import Styled from 'styled-components';
import { blue, darkGrey } from '../../Style/colors';

const StyledSearchBar = Styled.div `
  display: flex;
  flex-direction: row;
  border: 1px solid ${darkGrey};
  max-width: 500px;
  margin: 20px auto;
  border-radius: 4px;

  & input {
    flex-grow: 2;
    width: 70%;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    border-left: 0;
    border-bottom: 0;
    border-top: 0;
    border-right: 1px solid ${darkGrey};
  }

  & button {
    background: ${blue};
    color: white;
    width: 30%;
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
`;

export default StyledSearchBar;