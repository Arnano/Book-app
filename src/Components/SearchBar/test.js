import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SearchBar from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Search bar', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar onClick={() => {}}>Search</SearchBar>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<SearchBar onClick={() => {}}>Search</SearchBar>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
