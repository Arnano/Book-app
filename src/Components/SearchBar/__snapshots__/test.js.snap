// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa dTdotJ"
>
  <input
    className="sc-bwzfXH bnZZpQ"
    onBlur={undefined}
    onChange={undefined}
    onClick={undefined}
    placeholder=""
    type="text"
    value={undefined}
  />
  <button
    className="sc-htpNat gAYoDj"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
